tests/end_to_end/test_create_new_template.py

```diff
@@ -0,0 +1,188 @@
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    send_messages_button = page.get_by_role("link", name="Send messages")
+    # expect(send_messages_button).to_be_visible()
+    send_messages_button.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    create_template_button = page.get_by_role("button", name="New template")
+    expect(create_template_button).to_be_visible()
+    create_template_button.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    start_with_a_blank_template_radio = page.get_by_text("Start with a blank template")
+    expect(start_with_a_blank_template_radio).to_be_visible()
+    start_with_a_blank_template_radio.click()
+
+    continue_button = page.get_by_role("button", name="Continue")
+
+    # continue_button = page.get_by_text("Continue")
+    expect(continue_button).to_be_visible()
+    continue_button.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    template_name_input = page.get_by_text("Template name")
+    expect(template_name_input).to_be_visible()
+    template_name = str(uuid.uuid4())
+    template_name_input.fill(template_name)
+    message_input = page.get_by_role("textbox", name="Message")
+    expect(message_input).to_be_visible()
+    message = "Test message for e2e test"
+    message_input.fill(message)
+
+    save_button = page.get_by_text("Save")
+    expect(save_button).to_be_visible()
+    save_button.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    use_this_template_button = page.get_by_text("Use this template")
+    expect(use_this_template_button).to_be_visible()
+    use_this_template_button.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    use_my_phone_number_link = page.get_by_text("Use my phone number")
+    expect(use_my_phone_number_link).to_be_visible()
+    use_my_phone_number_link.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    preview_button = page.get_by_text("Preview")
+    expect(preview_button).to_be_visible()
+    preview_button.click()
+
+    # Check to make sure that we've arrived at the next page.
+    page.wait_for_load_state("domcontentloaded")
+
+    # We are not going to send the message for this test, we just want to confirm
+    # that the template has been created and we are now seeing the message from the
+    # template in the preview.
+    assert "Test message for e2e test" in page.content()
+
+
+def test_create_new_template(authenticated_page, end_to_end_context):
+    page = authenticated_page
+
+    _setup(page, end_to_end_context)
```

Comment:
See the note about in the `_setup()` method - I don't think we need `end_to_end_context` here at all and it could introduce other complications down the line.